### PR TITLE
[WIP] Setup structure to use Thor for the CLI

### PR DESCRIPTION
**Pull request by davidstosik**

### CODE REVIEW

File: lib/shinseibank/cli/subcommand.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class ShinseiBank
+  class CLI < Thor
+    class Subcommand < Thor
+      class_option :credentials, required: true, type: :string
+
+      private
+
+        def shinseibank
+          @_shinseibank ||= ShinseiBank.new
```
Comment (knshiro) [inline lib/shinseibank/cli/subcommand.rb]: why the underscore? it's to show it's private?

File: lib/shinseibank/cli/account.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require "ostruct"
+require "shinseibank/cli/subcommand"
+
+class ShinseiBank
+  class CLI < Thor
+    class Account < Subcommand
+      desc "show", "Show your account details"
+      def show
+        login
```
Comment (knshiro) [inline lib/shinseibank/cli/account.rb]: feels like this is going to be required for all entrypoints. (logout would be nice too)
Is there any way to wrap all calls between those 2?

File: lib/shinseibank/cli/subcommand.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class ShinseiBank
+  class CLI < Thor
+    class Subcommand < Thor
+      DEFAULT_CREDENTIALS_PATH = "./shinsei_bank.yaml".freeze
+
+      class_option :credentials, required: true, type: :string, aliases: "-c", default: DEFAULT_CREDENTIALS_PATH
```
Comment (knshiro) [inline lib/shinseibank/cli/subcommand.rb]: If it has a default it's not required :D

File: lib/shinseibank/cli/subcommand.rb (hunk 1 above)
Comment (davidstosik) [reply lib/shinseibank/cli/subcommand.rb]: Heh, yeah, it's become a habit at work: instance variables used as "cache", not supposed to be used anywhere else, are prefixed by an underscore. I don't have to use that "private rule" here, but I think it's nice.

File: lib/shinseibank/cli/account.rb (hunk 2 above)
Comment (davidstosik) [reply lib/shinseibank/cli/account.rb]: I'll think about something better.

File: lib/shinseibank/cli/subcommand.rb (hunk 3 above)
Comment (davidstosik) [reply lib/shinseibank/cli/subcommand.rb]: True.

File: lib/shinseibank/cli/account.rb (hunk 2 above)
Comment (davidstosik) [inline lib/shinseibank/cli/account.rb]: @ugo Didn't find anything nice... There's a Thor Hollaback gem, to attach before/after/around callbacks, but that feels a bit too much.

Comment (davidstosik) [reply lib/shinseibank/cli/account.rb]: @knshiro  Didn't find anything nice... There's a Thor Hollaback gem, to attach before/after/around callbacks, but that feels a bit too much.

**Comment by knshiro** [comment]
You are going in too many directions in this PR.
Let's merge it since it does create a base structure for the CLI and it only breaks the buy funds that we are not using so far.
What do you think?

**Comment by davidstosik** [comment]
@knshiro Heh, very true. 😅 
Or I can split if you want me to!
I guess we can merge that before working more on it. Any remark on the code?

**Comment by knshiro** [comment]
LGTM

**Pull request merged**